src/pages/ScholarshipForm.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { Container, TextField, Button } from '@material-ui/core';
+import { ScholarshipFields } from './formFields';
+
+function ScholarshipForm() {
+  const [formFieldStates, setFormFieldStates] = useState({
+    name: '',
+    deadline: new Date(),
+    description: '',
+    amountType: '',
+    amount: 0,
+    minAmount: 0,
+    maxAmount: 0,
+    website: '',
+    scholarshipType: '',
+    active: false,
+  });
+
+  function handleFieldChange(index, value) {
+    setFormFieldStates((prev) => ({ ...prev, [index]: value }));
+  }
+  function submitScholarship() {
+    console.log(formFieldStates); // eslint-disable-line no-console
+  }
+
+  // displays the fields for amount depending on the type of amount
+  function displayAmountFields(index) {
+    const amountTypeState = formFieldStates.amountType;
+    if (
+      !amountTypeState ||
+      amountTypeState === 'unknown' ||
+      amountTypeState === 'fullride'
+    )
+      return <div key="none" />;
+    if (amountTypeState === 'fixed') {
+      const rewardAmountRef = ScholarshipFields.rewardAmount.amount;
+      return (
+        <TextField
+          key={rewardAmountRef.key}
+          id="amount"
+          label={rewardAmountRef.label}
+          type="number"
+          InputLabelProps={{ shrink: true }}
+          fullWidth
+          InputProps={{ inputProps: { min: 0, step: 50 } }}
+          value={amountTypeState}
+          onChange={(e) =>
+            handleFieldChange(index, parseInt(e.target.value, 10) || '')
+          }
+        />
+      );
+    }
+    return ['minAmount', 'maxAmount'].map((field) => (
+      <TextField
+        key={ScholarshipFields.rewardAmount[field].key}
+        id={field}
+        label={ScholarshipFields.rewardAmount[field].label}
+        type="number"
+        InputLabelProps={{ shrink: true }}
+        fullWidth
+        InputProps={{ inputProps: { min: 0, step: 50 } }}
+        value={formFieldStates[field]}
+        onChange={(e) =>
+          handleFieldChange(field, parseInt(e.target.value, 10) || '')
+        }
+      />
+    ));
+  }
+
+  return (
+    <Container style={{ width: '40%' }}>
+      <h1>Submit a Scholarship</h1>
+      <form noValidate autoComplete="off">
+        {Object.entries(ScholarshipFields).map(
+          ([field, component]) => component
+        )}
```

Comment:
I tried that way and it didn't render anything. Maybe it's because I didn't forward the refs? Regardless, I'm trying it again and will let you know my status on it.

Reply:
What about
```jsx
        {Object.entries(ScholarshipFields).map(
          ([name, Field]) => <Field id={name} key={name}/>
        )}
```

If that doesn't work then I think it's because `field` is an object rather than a class. Hmm, in that case I would forget about this suggestion.